core/src/main/java/hudson/model/User.java

```diff
@@ -185,6 +190,8 @@ public class User extends AbstractModelObject implements AccessControlled, Descr
         XSTREAM.alias("user", User.class);
     }
 
+    private User() {}
```

Comment:
There is a subtle difference between calling the no-arg ctor and unmarshalling into that object, vs. deserializing a new object: whether `properties` (and, if anyone cares, `version`) are initialized.